spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewritePositionDeleteFilesSparkAction.java

```diff
@@ -140,6 +148,33 @@ public RewritePositionDeleteFiles.Result execute() {
     }
   }
 
+  private boolean hasDeletionVectors() {
```

Comment:
I don't feel like we're scanning for the right thing here.  Shouldn't we be checking to see if there are any positional delete files here (not puffin DVs)?

If this is run on a V3 table, it shouldn't matter whether there are already DV files, only that that there are positional delete files that need to be rewritten.

That said, this should probably be called something like `requiresDVRewrite()`